src/components/button/button.tsx

```diff
@@ -0,0 +1,73 @@
+import { ReactNode, useState } from "react";
+import { constantsButton } from ".";
+import { constants } from "../../typography";
+import { colors } from "../../typography/colors";
+
+const stylesButton = {
+  minWidth: constantsButton.minWidth,
+  fontFamily: constants.font.family.primary,
+  fontWeight: constantsButton.weight,
+  borderRadius: constants.font.radius,
+  cursor: constantsButton.cursor,
+};
+interface ButtonProps {
+  type?: "defaultCompact" | "disableCompact" | "disableDefault";

```

Comment:
`disabled` e `compact` são duas características separadas (ele ser `compact` não afeta nenhuma prop que é alterada se ele for `disabled`), vc pode fazer duas booleans ao invés de fazer um tipo com 4 strings